### ISSUE TITLE
Can't finish RUM setup

### ISSUE THREAD

**Issue by AlanSalas** [DataDog/dd-sdk-reactnative#340]
**Describe what happened**
Im trying to connect datadog dashboard to my app, i followed every single step in the docs and in the RUM section when create a new app.

This is my config file:

```
import { DdSdkReactNativeConfiguration } from '@datadog/mobile-react-native';

const config = new DdSdkReactNativeConfiguration(
  '...',
  'env',
  '...',
  true, // track User interactions
  true, // track XHR Resources
  true, // track Errors
);

config.site = 'US1';
config.nativeCrashReportEnabled = true;
config.sampleRate = 100;

export default config;
```
then i create a function to init datadog

```
import { DdSdkReactNative } from '@datadog/mobile-react-native';
import datadogConfig from './datadogConfig';

const datadogInit = () => {
  DdSdkReactNative.initialize(datadogConfig)
    .then(() => {
      console.log('Works');
      DdSdkReactNative.setUser({
          id: '123',
         name: 'test',
      });
    })
    .catch(err => {
      console.error(err);
    });
};

export default datadogInit;
```

then i call that function in my index, i build my app and run it and in the console it shows me the console.log("Works") but the dashboard is Waiting for data...

**Additional context**

-   Version of the SDK: 6.9
-   React native 0.66.4
-   @datadog/mobile-react-native 1.0.0-rc4

**Comment by louiszawadzki** [comment]
Hi @AlanSalas, thanks for reaching out!

We've just added a guide to help you troubleshoot this case: https://github.com/DataDog/dd-sdk-reactnative/blob/develop/docs/troubleshooting.md
Could you give it a try and tell me if you managed to solve your problem?
This is brand new so feel free to ask if anything is unclear or hard to achieve in your setup so we can improve it :) 

If not, I encourage you to [contact us through our support team](https://www.datadoghq.com/support/), that will make it easier to exchange logs and bring more people into the discussion.
You can mention this issue and the logs you obtained from the troubleshooting guide to make the resolution faster :) 

Thanks a lot!

Thanks a lot!

**Comment by AlanSalas** [comment]
Hi @louiszawadzki thanks for your response. I tried to use the import SdkVerbosity from '@datadog/mobile-react-native' to see the logs but shows me an error, look:

![image](https://user-images.githubusercontent.com/46615148/202765430-71337aac-96eb-41f0-8c42-df5b3805ea59.png)

but i can use the DdSdkReactNativeConfiguration for create the config 🤔 , im in @datadog/mobile-react-native 1.0.0-rc4 btw

**Comment by louiszawadzki** [comment]
Hi @AlanSalas, 

`SdkVerbosity` was only introduced in version `1.1.0-rc5`.
I would recommend you to upgrade to the latest version `1.1.4`, it should be working with your version of React Native.
Is there anything preventing you from upgrading that we can help on?

**Comment by AlanSalas** [comment]
Hi @louiszawadzki, the first time i tried to install the latest version `1.1.4` but it throws errors at the time of installation and I tried with previous versions, but let me try again and if I have any errors I'll show you

**Comment by AlanSalas** [comment]
Hello, I no longer have the ts error, in fact I can already use sdkVerbosity.DEBUG and I can see some logs in console, I show you: 

![image](https://user-images.githubusercontent.com/46615148/203603066-afa56619-b712-4866-a307-a0743ee6d2a7.png)
![image](https://user-images.githubusercontent.com/46615148/203603382-bb74ad07-b7dd-46b4-b53a-b588f5cac1ac.png)
![image](https://user-images.githubusercontent.com/46615148/203603424-dce2033c-1c41-4166-ae53-c674f1a48f69.png)

As you can see datadog is initializing, finally! I also did a test setUser but in the dashboard it keeps waiting for data...
I also configure the NavigationContainer component with DdRumReactNavigationTracking and it shows me an error 

![image](https://user-images.githubusercontent.com/46615148/203604922-af2fea07-b261-408c-aa91-28f1ba8212e3.png)

**Comment by louiszawadzki** [comment]
Hi again @AlanSalas,

This could be an issue with the react-navigation integration. If it does not work and does not create any Datadog view, then it's expected that nothing will show up as events won't be attached to any Datadog view.

Could you share which version of `react-navigation` you are using and how you are calling `DdRumReactNavigationTracking.startTrackingViews`?

We only support `react-navigation` v5 and v6 for information.

**Comment by AlanSalas** [comment]
Its weird because only at the beginning of my application does this warning appear, then in the datadog console it tells me when the view is changed, when there is an HTTP request, etc. In other words, it is already throwing events, only it does not reflect them on the dashboard...

React-navigation version: 6.0.8

And in my App.tsx

```
import { DdRumReactNavigationTracking } from '@datadog/mobile-react-navigation';

const App = () => {
  const navigationRef = useRef(null);

  useEffect(() => {
    setTimeout(() => {
      SplashScreen.hide();
    }, 1000);
  }, []);

  return (
    <React.StrictMode>
      <ErrorBoundary>
        <AppProviders>
          <NavigationContainer
            ref={navigationRef}
            onReady={() => {
              DdRumReactNavigationTracking.startTrackingViews(navigationRef.current);
            }}
          >
            <Navigator />
          </NavigationContainer>
        </AppProviders>
      </ErrorBoundary>
    </React.StrictMode>
  );
};

export default App;

```

**Comment by AlanSalas** [comment]
Hello again, apparently the connection to the dashboard happened and it already gives me information related to my application, thank you very much, I just have one more question, why when adding the environment variables and calling them in the datadog configuration does this error appear?

![image](https://user-images.githubusercontent.com/46615148/203848305-d214fa4a-3672-4733-bc2c-739cbd687e6f.png)

any suggestions?

**Comment by louiszawadzki** [comment]
Hi @AlanSalas,
This error indicates there is an issue with your environment name (`DATADOG_ENV_NAME` in your code). 

The React Native SDK uses the Android SDK under the hood, and this SDK does validate the environment name is correctly formatted for Datadog. You can see the [validation function here](https://github.com/DataDog/dd-sdk-android/blob/develop/dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt#L367), and the [Regex it uses here](https://github.com/DataDog/dd-sdk-android/blob/develop/dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt#L450). 

What is the real value you use for the environment?

**Comment by AlanSalas** [comment]
The real value is frqtal :/ i dont see nothing bad with the name

**Comment by louiszawadzki** [comment]
Hi @AlanSalas,

Did you check that `react-native-config` is properly installed and this env variable resolves to its intended value in the js code?
There is an [extra installation step for Android](https://github.com/luggit/react-native-config#extra-step-for-android) that modifies `android/app/build.gradle`, and if I remove this part I get an undefined value which leads to the same error you have:
```
// 2nd line, add a new apply:
apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"
```

**Comment by louiszawadzki** [comment]
Hi @AlanSalas, 

I hope that you found the solution to your issue! As it's not been active for a while I'm going to close it, feel free to reopen it if this is not the case!

**Comment by elmcapp** [comment]
I still have this issue

**Comment by louiszawadzki** [comment]
Hi @elmcapp,

Could you provide more information as to what you are experiencing? 
If your use case is the same as the one described in the issue, could you specify if you checked that `react-native-config` is properly installed and this env variable resolves to its intended value in the js code?
There is an [extra installation step for Android](https://github.com/luggit/react-native-config#extra-step-for-android) that modifies `android/app/build.gradle`, and if I remove this part I get an undefined value which leads to the same error you have:

```
// 2nd line, add a new apply:
apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"
```

If this is the case, can you provide us with a small reproducible app to help us assist you in your troubleshooting?

If you are in a different use case, please feel free to open a new issue :) 

Thanks a lot!
